File: aoc-2024/app/src/test/kotlin/com/github/leomartins1999/aoc/day7/Day7Test.kt
package com.github.leomartins1999.aoc.day7

import kotlin.test.Test
import kotlin.test.assertEquals

class Day7Test {

    private val input = """
        190: 10 19
        3267: 81 40 27
        83: 17 5
        156: 15 6
        7290: 6 8 6 15
        161011: 16 10 13
        192: 17 8 14
        21037: 9 7 18 13
        292: 11 6 16 20
    """.trimIndent()

    @Test
    fun part1Example() {
        assertEquals(3749, Day7(input).part1())
    }
}

File: aoc-2025/day7/day_7_test.go
package Day7_test

import (
	Day7 "aoc-2025/day7"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_SolvePart1(t *testing.T) {
	day := &Day7.Day7{InputFilePath: "./day-7-test.txt"}

	assert.Equal(t, 21, day.SolvePart1())
}

func Test_SolvePart2(t *testing.T) {
	day := &Day7.Day7{InputFilePath: "./day-7-test.txt"}

	assert.Equal(t, 40, day.SolvePart2())
}
